packages/@ember/-internals/glimmer/lib/utils/string.ts

```diff
@@ -43,13 +43,13 @@ export function escapeExpression(string: any): string {
     } else if (string === null || string === undefined) {
       return '';
     } else if (!string) {
-      return string + '';
+      return String(string);
```

Comment:
@rwjblue unless I interpret the MDN docs incorrectly it will only return a boxed version if we use `new String(...)`, but `String(...)` will only cast the value

Reply:
Ahh, I see. Doesn't that actually make it more likely to make mistakes though (generally speaking `String()` vs `new String()` pretty subtle to me)?

Reply:
we could enable https://eslint.org/docs/rules/no-new-wrappers to prevent that. not sure if we're using it anywhere for good reasons though.

Reply:
Sounds good, lets put those together 👍

Reply:
done